src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,17 +29,24 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        String[] addressParts = trimmedAddress.split(", "); //split the address into sections
+        if (!isValidAddress(addressParts)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        // Set each section of the address
+        // Also checks if each section is valid
+        setBlock(new AddressBlock(addressParts[0]));
```

Comment:
You seem to have the intention to apply SLAP?  Notice that at this level of abstraction, the method call does not seem to go well with the rest of the code which is of a different level of abstraction.  In fact, a simple `this.AddressBlock(addressParts[0])` will suffice here.  Remember: SLAP is not only about high level, but also about same level.